zarr/tests/test_convenience.py

```diff
@@ -229,64 +236,81 @@ def test_consolidate_metadata(with_chunk_store, zarr_version):
     arr[:] = 1.0
     assert 16 == arr.nchunks_initialized
 
+    if stores_from_path:
```

Comment:
Seems fine to have in the test, but this is an interesting block, @grlee77. Can you explain it?

Reply:
There are tests below that do `del store[key]`, but when the store is just a path string instead of a `BaseStore` class this will not work. So, the test uses `z._store` directly. I think I had originally just tried deleting the keys directly from the consolidated store, `z`, instead of accessing the private `z._store`, but this was not possible because `ConsolidatedMetadataStore` classes are read-only.